app/models/order.rb

```diff
@@ -0,0 +1,67 @@
+class Order < ApplicationRecord
+  belongs_to :cart
+  enum status: [:ordered,:completed,:cancelled,:paid]
+  TYPE={ordered:"ordered",completed:"completed",cancelled:"cancelled",paid:"paid",all:"all"}
+
+  after_initialize do
+    if self.new_record?
+      self.status ||= :ordered
+    end
+  end
+
+  def self.recent_orders(current_user)
+    self.where(user_id:current_user.id).order("created_at DESC")
+  end
+
+
+
+
+  def self.place_order(cart)
+    total=self.total_order_price(cart)
+    order=self.new(cart_id:cart.id,user_id:cart.user_id,total_amount:total)
+    if order.save!
+      order
```

Comment:
I think you can return `order.save` and direclty make your decision in controller.

Reply:
I need order object in my controller if the order is saved, therefore I did not return _order.save_  .